api/app/src/providers/dexcom.ts

```diff
@@ -0,0 +1,190 @@
+  static AUTHORIZATION_PATH = "/v2/oauth2/login";
+  static TOKEN_PATH = "/v2/oauth2/token";
+  static scopes = "offline_access";
+
+  private static clientId = Config.getDexcomClientId();
+  private static clientSecret = Config.getDexcomClientSecret();
+
+  constructor(
+    private readonly oauth = new OAuth2DefaultImpl(
+      PROVIDER_DEXCOM,
+      Dexcom.clientId,
+      Dexcom.clientSecret,
+      {
+        authorizeHost: Dexcom.URL,
+        tokenHost: Dexcom.URL,
+        authorizePath: Dexcom.AUTHORIZATION_PATH,
+        tokenPath: Dexcom.TOKEN_PATH,
+      },
+      Dexcom.scopes
+    )
+  ) {
+    super({
+      [ConsumerHealthDataType.Activity]: false,
+      [ConsumerHealthDataType.Body]: false,
+      [ConsumerHealthDataType.Biometrics]: true,
+      [ConsumerHealthDataType.Nutrition]: true,
+      [ConsumerHealthDataType.Sleep]: false,
+      [ConsumerHealthDataType.User]: false,
+    });
+  }
+
+  private async getAccessToken(connectedUser: ConnectedUser): Promise<string> {
+    const providerData = getProviderDataFromConnectUserOrFail(connectedUser, PROVIDER_DEXCOM);
+
+    const token = providerData.token;
+
+    const refreshedToken = await this.checkRefreshToken(token, connectedUser);
+
+    return refreshedToken.access_token;
+  }
+
+  async checkRefreshToken(token: string, connectedUser: ConnectedUser): Promise<Token> {
+    const access_token = JSON.parse(token);
+    const bufferSeconds = 600;
+    const isExpired =
+      convert(access_token.expires_at).from("s").to("ms") -
+        Date.now() +
+        convert(bufferSeconds).from("s").to("ms") <=
+      0;
+
+    if (isExpired) {
+      const formData = {
+        grant_type: "refresh_token",
+        refresh_token: access_token.refresh_token,
+        redirect_uri: this.oauth.getRedirectUri(),
+        client_id: Dexcom.clientId,
+        client_secret: Dexcom.clientSecret,
+      };
+
+      const formBody = new URLSearchParams(formData).toString();
+
+      try {
+        const response = await axios.post(`${Dexcom.URL}${Dexcom.TOKEN_PATH}`, formBody, {
+          headers: {
+            "Content-Type": "application/x-www-form-urlencoded",
+          },
+        });
+
+        response.data.expires_at = dayjs().add(response.data.expires_in, "seconds").unix();
+
+        const providerItem = connectedUser.providerMap
+          ? {
+              ...connectedUser.providerMap[PROVIDER_DEXCOM],
+              token: JSON.stringify(response.data),
+            }
+          : { token: JSON.stringify(response.data) };
+
+        await updateProviderData({
+          id: connectedUser.id,
+          cxId: connectedUser.cxId,
+          provider: PROVIDER_DEXCOM,
+          providerItem,
+        });
+
+        return response.data;
+      } catch (error) {
+        console.log("Error refreshing access token: ", error);
+        capture.error(error, {
+          extra: {
+            context: `dexcom.refreshToken`,
+            cxId: connectedUser.cxId,
```

Comment:
Not sure if this is the case, but if we're calling `.getCxIdXXX()` on the route level, we don't need to set the `cxId` on Sentry's context as its already done there (as `user.id`).

If we're not and want to do it here, can we set as `user.id` like `.getCxId()` does, please?